src/System.Data.Common/tests/System/Data/Common/DbDataReaderMock.cs

```diff
@@ -41,12 +40,7 @@ public DbDataReaderMock()
 
         public DbDataReaderMock(DataTable testData)
         {
-            if (testData == null)
-            {
-                throw new ArgumentNullException(nameof(testData));
-            }
-
-            _testDataTable = testData;
+            _testDataTable = testData ?? throw new ArgumentNullException(nameof(testData));
```

Comment:
I'd rather this PR was constrained to the method overloads we're adding, without modernizing or doing anything beyond that - let's separate API changes from internal cleanup. But it would definitely be good to have these cleanups (as well as some others I'm seeing) in a separate PR after this one.

Reply:
Reverted.